File: client/src/app/shared/models/announcementDto.ts
import {Campground} from "./campground";

export interface AnnouncementDto {
    id: number;
    title: string;
    subtitle: string | null;
    content: string;
    expirationDate: string | null;
    messageType: string;
    forceGlobal: boolean;
    pinnedPriority: number | null;
    createdAt: Date;
    createdBy: string | null;
    updatedAt: Date | null;
    updatedBy: string | null;
    campgrounds: Campground | null;
}

File: API/DTOs/AnnouncementDto.cs
namespace API.DTOs;

public class AnnouncementDto
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public string? Subtitle { get; set; }
    public required string Content { get; set; }
    public DateTime? ExpirationDate { get; set; }
    public required string MessageType { get; set; }
    public bool ForceGlobal { get; set; }
    public int? PinnedPriority { get; set; }
    public DateTime CreatedAt { get; set; }
    public required string CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }
    public ICollection<CampgroundDto>? Campgrounds { get; set; }
}